File: 단계별로 풀어보기/기본 수학1/2292/벌집.cpp
#include <iostream>
using namespace std;
//6의 배수 누적 합

int findNumOfRoom(int N){
    int total = 0;
    int i;
    
    if(N==1)
        return 1;
        
    for(i=0; !((total<N)&&(N<=total+i*6+1));i++){
        total+=i*6;
    }
    
    
    return i+1;
}


int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    
    int N;
    cin >> N;
    
    cout<<findNumOfRoom(N);

    return 0;
}

File: 단계별로 풀어보기/기본 수학1/2292/벌집.py
import sys
def findNumOfRoom(N):
    total = 0
    i=0    
    while(not((total<N)and(N<=total+i*6+1))):
        total+=i*6
        i+=1
    
    return i+1
    
def init():
    N=int(sys.stdin.readline())
    print(findNumOfRoom(N))
    
init()
